File: WarnIt-Address_old/src/main/java/br/com/warnit/address/repository/EstadoRepository.java
package br.com.warnit.address.repository;

import br.com.warnit.address.model.domain.Estado;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

/**
 * @author Guilherme Camargo
 * @since 01/02/2019
 * @version 1.0
 * */
@Repository
public interface EstadoRepository extends JpaRepository<Estado, Long> {

    @Transactional(readOnly = true)
    Optional<Estado> findByNomeEstado(String nomeEstado);
}

File: WarnIt-Address/src/main/kotlin/br/com/warnit/address/repository/EstadoRepository.kt
package br.com.warnit.address.repository

import org.springframework.data.repository.CrudRepository
import br.com.warnit.address.model.Estado
import org.springframework.transaction.annotation.Transactional


interface EstadoRepository: CrudRepository<Estado, Long>{
    @Transactional(readOnly = true)
    fun findByNome(nome: String): Estado
}